extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ParquetFileWriter.java

```diff
@@ -54,67 +51,70 @@ public ParquetFileWriter(
         this.targetPageSize = targetPageSize;
         this.allocator = allocator;
         this.extraMetaData = new HashMap<>(extraMetaData);
-        writeChannel = channelsProvider.getWriteChannel(filePath, false); // TODO add support for appending
-        writeChannel.write(ByteBuffer.wrap(ParquetFileReader.MAGIC));
+        // TODO add support for appending
+        bufferedOutput = new BufferedStreamOverWriteChannel(channelsProvider.getWriteChannel(filePath, false),
+                OUTPUT_BUFFER_SIZE);
+        bufferedOutput.write(ParquetFileReader.MAGIC);
         this.type = type;
         this.channelsProvider = channelsProvider;
         this.compressorAdapter = DeephavenCompressorAdapterFactory.getInstance().getByName(codecName);
     }
 
+    /**
+     * Following method is unused and untested.
+     */
     @SuppressWarnings("unused")
     RowGroupWriter addRowGroup(final String path, final boolean append) throws IOException {
-        RowGroupWriterImpl rowGroupWriter =
-                new RowGroupWriterImpl(path, append, channelsProvider, type, targetPageSize, allocator,
-                        compressorAdapter);
+        bufferedOutput.close();
+        bufferedOutput =
+                new BufferedStreamOverWriteChannel(channelsProvider.getWriteChannel(path, append), OUTPUT_BUFFER_SIZE);
```

Comment:
This isn't the appropriate thing to do. You should pass a new channel to the new writer.

Reply:
This is what was happening earlier as well in the file RowGroupWriterImpl.java [here]( https://github.com/deephaven/deephaven-core/pull/4669/files/50dfd20d8595efaf3783583690aaa2a28a4f5a7c#diff-cbfe79bd13d25860ebfbea418157f838ea092c5562ff57fa82f7ad74df5296b0L40:~:text=this(channelsProvider.getWriteChannel(path%2C%20append)%2C%20type%2C%20targetPageSize%2C%20allocator%2C%20blockWithPath(path)%2C ). I just moved it up one layer.
Do you want me to change it?

I added a close in line 68 so that we close and flush the old channel and create a new one.